server/src/test/scala/com/ossum/riddl/lsp/server/RiddlLSPTextDocumentSpec.scala

```diff
@@ -0,0 +1,197 @@
+    val changeNotification: DidChangeTextDocumentParams =
+      new DidChangeTextDocumentParams()
+
+    val versionedDocIdentifier = new VersionedTextDocumentIdentifier()
+    versionedDocIdentifier.setUri(docURI)
+    changeNotification.setTextDocument(versionedDocIdentifier)
+
+    val changes = new TextDocumentContentChangeEvent()
+    changes.setText("the")
+    val changeRange = new lsp4j.Range()
+
+    val changeStart = new Position()
+    changeStart.setLine(11)
+    changeStart.setCharacter(21)
+    changeRange.setStart(changeStart)
+
+    val changeEnd = new Position()
+    changeEnd.setLine(11)
+    changeEnd.setCharacter(24)
+    changeRange.setEnd(changeEnd)
+
+    changes.setRange(changeRange)
+    changeNotification.setContentChanges(List(changes).asJava)
+
+    service.didChange(changeNotification)
+  }
+
+  trait ChangeEmptyFileSpec extends OpenEmptyFileSpec with OneErrorFileSpec {
+    val changeNotification: DidChangeTextDocumentParams =
+      new DidChangeTextDocumentParams()
+
+    val versionedDocIdentifier = new VersionedTextDocumentIdentifier()
+    versionedDocIdentifier.setUri(docURI)
+    changeNotification.setTextDocument(versionedDocIdentifier)
+
+    val changes = new TextDocumentContentChangeEvent()
+    changes.setText("domain New {}")
+    val changeRange = new lsp4j.Range()
+
+    val changeStart = new Position()
+    changeStart.setLine(1)
+    changeStart.setCharacter(1)
+    changeRange.setStart(changeStart)
+
+    val changeEnd = new Position()
+    changeEnd.setLine(1)
+    changeEnd.setCharacter(1)
+    changeRange.setEnd(changeEnd)
+
+    changes.setRange(changeRange)
+    changeNotification.setContentChanges(List(changes).asJava)
+
+    service.didChange(changeNotification)
+  }
+
+  "RiddlLSPTextDocumentService" must {
+    "successfully open everythingOneError.riddl & get completion" in new OpenOneErrorFileSpec {
+      val params = new CompletionParams()
+      params.setTextDocument(textDocumentIdentifier)
+      val position = new Position()
+      position.setLine(5)
+      position.setCharacter(7)
+      params.setPosition(position)
+
+      val resultF: CompletableFuture[
+        messages.Either[util.List[CompletionItem], CompletionList]
+      ] = service.completion(params)
+
+      whenReady(resultF.asScala) { eitherList =>
+        eitherList.isRight mustBe true
+        eitherList.getRight.getItems.asScala.length mustEqual 1
+        eitherList.getRight.getItems.asScala.head.getTextEditText mustEqual
+          """Expected one of (end-of-input | whitespace after keyword)"""
+      }
+    }
+
+    "successfully close everythingOneError.riddl" in new OpenOneErrorFileSpec {
+      val closeNotification: DidCloseTextDocumentParams =
+        new DidCloseTextDocumentParams()
+      closeNotification.setTextDocument(textDocumentIdentifier)
+      service.didClose(closeNotification)
+    }
+
+    "successfully change everythingOneError.riddl" in new ChangeOneErrorFileSpec {}
```

Comment:
This test case needs to be written.  You should generally write "pending" inside that `{}` to mark the test as pending completion, whenever you do this.